nifi-bootstrap/src/test/java/org/apache/nifi/bootstrap/util/TestSecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,208 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class TestSecureNiFiConfigUtil {
+    private Logger logger = LoggerFactory.getLogger("org.apache.nifi.bootstrap.util.TestSecureNiFiConfigUtil");
+
+    private static final boolean EXPECT_STORES_TO_EXIST = true;
+
+    private String nifiPropertiesFile = "src/test/resources/nifi.properties";
+    private String keystorePath = "src/test/resources/keystore.p12";
+    private String truststorePath = "src/test/resources/truststore.p12";
+
+    private NiFiProperties configureSecureNiFiProperties(String testPropertiesFile) throws IOException {
+        Files.copy(new File(testPropertiesFile).toPath(), new File(nifiPropertiesFile).toPath(), StandardCopyOption.REPLACE_EXISTING);
+        SecureNiFiConfigUtil.configureSecureNiFiProperties(nifiPropertiesFile, logger);
+
+        return new NiFiPropertiesLoader().load(nifiPropertiesFile);
+    }
+
+    private static String getFileHash(String filename) throws NoSuchAlgorithmException, IOException {
+        MessageDigest md = MessageDigest.getInstance("MD5");
+        md.update(Files.readAllBytes(Paths.get(filename)));
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest).toUpperCase();
+    }
+
+    @After
+    public void cleanUp() throws IOException {
+        Files.deleteIfExists(new File(nifiPropertiesFile).toPath());
+        Files.deleteIfExists(new File(keystorePath).toPath());
+        Files.deleteIfExists(new File(truststorePath).toPath());
+    }
+
+    private void runTestWithExpectedSuccess(String testPropertiesFile, List<String> expectedSANs) throws IOException, TlsException, KeyStoreException,
+            NoSuchAlgorithmException, UnrecoverableKeyException, CertificateException, InvalidKeyException, NoSuchProviderException, SignatureException {
+        NiFiProperties niFiProperties = this.configureSecureNiFiProperties(testPropertiesFile);
+
+        keystorePath = niFiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE);
+        Assert.assertEquals("./src/test/resources/keystore.p12", keystorePath);
+        Assert.assertFalse(niFiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_PASSWD).isEmpty());
+        Assert.assertEquals("PKCS12", niFiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_TYPE));
+
+        char[] keyPassword = niFiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_PASSWD).toCharArray();
+        KeyStore keyStore = KeyStoreUtils.loadKeyStore(keystorePath,
+                niFiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_PASSWD).toCharArray(),
+                niFiProperties.getProperty(NiFiProperties.SECURITY_KEYSTORE_TYPE));
+        String alias = keyStore.aliases().nextElement();
+        Assert.assertTrue(keyStore.isKeyEntry(alias));
+        Key key = keyStore.getKey(alias, keyPassword);
+        Assert.assertEquals("PKCS#8", key.getFormat());
```

Comment:
Is there a reason for checking the key format?

Reply:
Just to show that it did indeed generate the expected format.  But I suppose it already asserts the keystore type is PKCS12, and the test would already fail if it tried to read that format and it wasn't actually PKCS12.